File: project-euler/problem02.js
let fibonacci = [1, 2]
let last = fibonacci[fibonacci.length - 1]
while(last < 4000000){
 last = fibonacci[fibonacci.length - 1]
 let next = fibonacci[fibonacci.length - 2]
 fibonacci.push(last + next)
}
let even = fibonacci
.filter(num => num % 2 === 0)
.reduce((a, b) => a + b)
console.log(even)

File: project-euler/problem02.py
import functools

fibonacci = [1, 2]
last = fibonacci[len(fibonacci) -1]
while last < 4000000:
    last = fibonacci[len(fibonacci) -1]
    next = fibonacci[len(fibonacci)-2]
    fibonacci.append(next + last)
evens = [i for i in fibonacci if i % 2 == 0]
even_sum = functools.reduce(lambda a, b: a + b, evens)
print(even_sum)
